File: TFLiteARCore/app/src/main/java/com/ml/objectdetection/Coco.java
package com.ml.objectdetection;

import java.util.ArrayList;
import java.util.List;

import android.content.res.AssetManager;
import android.graphics.RectF;

import com.common.Logger;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.ByteBuffer;
import java.nio.ByteOrder;

public class Coco
{
    private static final Logger LOGGER = new Logger();
    public List<Image> images;
    public Coco(final AssetManager assetManager,final String filename)throws IOException
    {
        images = new ArrayList<Image>();

        //String actualFilename = filename.split("file:///android_asset/")[1];
        InputStream labelsInput = assetManager.open(filename);
        BufferedReader br = new BufferedReader(new InputStreamReader(labelsInput));
        String line;
        while ((line = br.readLine()) != null) {
            LOGGER.w(line);
            this.images.add(new Image(line));
        }
        br.close();
    }
    public class Image
    {
        public Image(String imageString)
        {
            String[] image = imageString.split(" ");
            this.path = image[0];
            this.labels = new ArrayList<Label>();
            for (int i = 1; i < image.length; i++)
            {
                labels.add(new Label(image[i]));
            }
        }
        public String path;
        public List<Label> labels;
        public class Label
        {
            public Label(String labelString)
            {
                String[] label = labelString.split(",");
                this.rect = new RectF();
                this.rect.left = Integer.parseInt(label[0]);
                this.rect.top = Integer.parseInt(label[1]);
                this.rect.right = Integer.parseInt(label[2]);
                this.rect.bottom = Integer.parseInt(label[3]);
                this.classIndex = Integer.parseInt(label[4]);
            }
            public RectF rect;
            public int classIndex;
            public boolean detected;
        }
    }
}

File: UnityBarracudaARFoundation/Assets/--MyProject--/Scripts/Objectdetection/Coco.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace ObjectDetection
{
    public class Coco
    {
        public List<Image> images;
        public Coco(TextAsset annotations)
        {
            images = new List<Image>();
            string text = annotations.text;
            string[] entries = text.Split('\n');
            foreach (string entry in entries)
            {
                this.images.Add(new Image(entry));
            }
        }
        public class Image
        {
            public Image(string imageString)
            {
                string[] image = imageString.Split(' ');
                this.path = image[0];
                this.labels = new List<Label>();
                for (int i = 1; i < image.Length; i++)
                {
                    labels.Add(new Label(image[i]));
                }
            }
            public string path;
            public List<Label> labels;
            public class Label
            {
                public Label(string labelString)
                {
                    string[] label = labelString.Split(',');
                    this.rect = new Rect();
                    int value = 0;
                    int.TryParse(label[0], out value);
                    this.rect.xMin = value;
                    int.TryParse(label[1], out value);
                    this.rect.yMin = value;
                    int.TryParse(label[2], out value);
                    this.rect.xMax = value;
                    int.TryParse(label[3], out value);
                    this.rect.yMax = value;
                    int.TryParse(label[4], out value);
                    this.classIndex = value;
                }
                public Rect rect;
                public int classIndex;
            }
        }
    }
}
